common/src/types/v0/store/switchover.rs

```diff
@@ -0,0 +1,156 @@
+use crate::types::v0::{
+    store::{
+        definitions::{ObjectKey, StorableObject, StorableObjectType},
+        SpecTransaction,
+    },
+    transport::{NexusId, VolumeId},
+};
+
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+
+/// Defines operation for SwitchOverSpec
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub enum Operation {
+    Init,
+    ShutdownOrigin,
+    ReconstructNexus,
+    SwitchOverNexus,
+    DeleteNexus,
+    PublishPath,
+    Errored(String),
+}
+
+/// Represent the state for the operation
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct OperationState {
+    operation: Operation,
+    result: Option<bool>,
+}
+
+impl OperationState {
+    pub fn new(operation: Operation, result: Option<bool>) -> Self {
+        Self { operation, result }
+    }
+}
+
+/// Defines timestamp for switchoverspec
+pub type SwitchOverTime = DateTime<Utc>;
+
+/// Represent switchover spec
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct SwitchOverSpec {
+    pub callback_uri: String,
+    pub old_nexus: Option<NexusId>,
+    pub new_nexus: Option<NexusId>,
```

Comment:
Don't think we should specify the nexusId here, that's internal to control plane.